functions.php

```diff
@@ -54,3 +56,71 @@ function displayDB(array $results) : string {
     return $dbResult;
 }
 
+
+/**
+ * Cleanse user input using filter_var() and return an array
+ *
+ * @param array $userInput user input in form is passed through
+ *
+ * @return array filtered user input collected in array and returned
+ */
+function cleanseData(array $userInput) : array {
+    $outputArr = [];
+    foreach($userInput as $strings){
+        if(!is_string($strings)){
+            return ['Incorrect input'];
+        }
+    }
+
+    foreach($userInput as $input){
+        $outputArr[] .= filter_var($input, FILTER_SANITIZE_SPECIAL_CHARS);
+    }
+    return $outputArr;
+}
+
+
+/**
+ * Generates a new DB link and inserts form data into db
+ *
+ * @param object $db this is the db connection we have established
+ * @param string $postArtist will take artists name from form
+ * @param string $postYearMade will take year painting made from form
+ * @param string $postPaintingName will take name of painting from form
+ * @param string $postImageLink will take url of image from form
+ */
+function sendFormData(object $db, string $postArtist, string $postYearMade, string $postPaintingName, string $postImageLink)
+{
+    $query = $db->prepare("INSERT INTO `collection-items` (`artist`, `year-made`, `painting-name`, `image-link`) 
+    VALUES (:artistName, :yearMade, :paintingName, :imageLink); ");
+
+    $query->bindParam(':artistName', $postArtist);
+    $query->bindParam(':yearMade', $postYearMade);
+    $query->bindParam(':paintingName', $postPaintingName);
+    $query->bindParam(':imageLink', $postImageLink);
+
+    $query->execute();
+}
+
+
+/**
+ * Using established db connection will query db to check if name of added painter is present in db,
+ * if name present in db, will send user to index.php with success message
+ * if name not present in db, will send user back to form page with error message
+ *
+ * @param object $db db connection
+ * @param string $postArtist artist name to check against db
+ */
+function getSubmission(object $db, string $postArtist){
+    $query = $db->prepare("SELECT `artist`, `year-made`, `painting-name`, `image-link` FROM `collection-items` 
+WHERE `artist` = :artistName ;");
+
+    $query->bindParam(':artistName', $postArtist);
+    $query->execute();
+    $results = $query->fetchAll();
+
+    if($results){
```

Comment:
probably best to return a boolean from this function based on presence of data or not then move the header lines to out side of the functions - perhaps with your procedural code at the top of form.php